enhancements/testing/openshift-tests-extension.md

```diff
@@ -0,0 +1,854 @@
+              },
+                
+              # Tests that require significant resources can identify
+              # that fact so that the planning algorithm will not
+              # overdraw on allocatable pod resources for a set of
+              # tests run in parallel.
+              "memory": "1Gi",
+              
+              # If a duration can be roughly predicted, inclusion of this
+              # information may improve the test execution 
+              # planning algorithm performance.  
+              "duration": "2s",
+                
+              # Timeout information may also support efficient
+              # planning.
+              "timeout": "16s",
+            },
+            
+            
+        }  
+    ],
+    
+    # Additional suites the extension wants to advertise / participate in.
+    "suites": [
+        {
+            # Here, the extension is advertising a new suite, unknown
+            # to the origin framework until discovered through the
+            # binary. 
+            "name": "fips/conformance",
+
+            "parents": [
+
+                # This suite can be run separately or will be included
+                # automatically in other suites known to origin.
+                # Here, the extension is informing origin that the suite
+                # is a sub-suite of openshift/conformance. All information
+                # about a suite is additive across extension binaries.
+                # For example, multiple binaries can advertise the same
+                # suite -- if they advertise different parents, then
+                # origin will treat the suite as a subset of each 
+                # identified parent.
+                "openshift/conformance"
+            ],
+
+            # Test cases can be advertised from across a number
+            # of extension binaries. The following cel expressions
+            # are OR'd together. If they select a test advertised
+            # from another binary, they will be considered part of
+            # this advertised suite.
+            # Test tags, names, and other attributes will be
+            # described later.
+            "qualifiers": [
+                "(test.tags.suite==\"fips\" || test.name.contains(\"fips\")) && !test.labels.has(\"Disruptive\")",
+                # Normally, CEL expressions evaluate against ALL test
+                # specs discovered by origin, regardless of the extension
+                # that advertised them. This makes it easy for 
+                # a suite defined in one extension to "absorb" tests 
+                # specified by another extension. To limit the expression
+                # to only tests in this extension, "owner" will be true
+                # only when evaluating tests advertised from this 
+                # extension.
+                # "owner" is equivalent to "extension.component == self.component"
+                "owner && test.name.contains(\"FIPS\"))"

```

Comment:
I'm not sure how to wire up the `owner` boolean value without a lot of complexity from origin.  Could we use `source` instead (equal to the component identifier)?

Example:

```golang
	ext.AddSuite(
		e.Suite{
			Name:    "example/tests",
			Parents: []string{"openshift/conformance/parallel"},
			Qualifiers: []string{
				`source == "openshift:payload:example-tests" `,
			},
		})
```

It also has the benefit of allowing opting in to some component's tests, but not all.

Reply:
The intent here was simplicity for the suite-writer. You solved this with the AddSuite vs AddGlobalSuite, so we drop 'owner'.